packages/transformer/src/BigMap.ts

```diff
@@ -0,0 +1,74 @@
+const MAX_MAP_SIZE = 10_000_000;
+
+export class BigMap<K, V> extends Map<K, V> {
+  private _maps: Map<K, V>[];
+  private _size: number;
+
+  public override get size(): number {
+    return this._size;
+  }
+
+  public constructor() {
+    super();
+    this._maps = [new Map()];
+    this._size = 0;
+  }
+
+  public override clear(): void {
+    this._maps.forEach((m) => m.clear());
+    this._size = 0;
+  }
+
+  public override delete(key: K): boolean {
+    const wasDeleted = this._maps.some((m) => m.delete(key));
+    if (wasDeleted) {
+      this._size--;
+    }
+
+    return wasDeleted;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public override forEach(callbackfn: (value: V, key: K, map: Map<K, V>) => void, thisArg?: any): void {
+    this._maps.forEach((m) => {
+      m.forEach(callbackfn, thisArg);
+    });
+  }
+
+  public override get(key: K): V | undefined {
+    for (const map of this._maps) {
+      const value = map.get(key);
+      if (value !== undefined) {
+        return value;
+      }
+    }
+
+    return;
+  }
+
+  public override has(key: K): boolean {
+    return this._maps.some((m) => m.has(key));
+  }
+
+  public override set(key: K, value: V): this {
```

Comment:
Would it be faster (and support approximately 167M elements) to have a submap bucket per last digit/character of the id64 during? Doesn't support as large of a list but will have O(1) access of buckets

Reply:
It will only be faster if we are going to have 20+ million of aspects in one iModel. If there are <10 million aspects, the set method will only call `has` once. By checking the last digit/character of the id64, it will be more of an overhead than adding another `has` invocation.

The optimization that you have suggested, will only make sense for extremely large datasets and I think it would have a performance hit in common scenarios.

Do you think it is still worth to change it?

Reply:
I think it will still be faster because it's less complicated than the iterator logic here even when the iterator contains only one element. I think we should try it to see if it's better than the 2-3% performance hit in all transforms. Another option is make this class use a regular map until the count hits 16.7M elements, then add another and start using iterations.